Mathlib/CategoryTheory/Limits/Set.lean

```diff
@@ -0,0 +1,45 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Limits.Lattice
+import Mathlib.CategoryTheory.Limits.Preserves.Basic
+import Mathlib.CategoryTheory.Limits.TypesFiltered
+
+/-!
+# The functor from `Set X` to types preserves filtered colimits
+
+Given `X : Type u`, the functor `Set.toTypes : Set X ⥤ Type u`
+which sends `A : Set X` to its underlying type preserves filtered colimits.
+
+-/
+
+universe w w' u
+
+open CategoryTheory Limits
+
+namespace Set
+
+/-- Given `X : Type u`, this the functor `Set X ⥤ Type u` which sends `A`
+to its underlying type. -/
+@[simps obj map]
+def toTypes {X : Type u} : Set X ⥤ Type u where
```

Comment:
I think the name `Set.toTypes` is too generic for the "global" `Set` namespace; it should be clear that this is a category theory thing, possibly containing "functor" somewhere in the name.

Also, it would be good if we could find a more discoverable place for this functor. Sadly, there seems to be no natural file that combines the required imports
```
import Mathlib.CategoryTheory.Category.Preorder
import Mathlib.CategoryTheory.Types
import Mathlib.Data.Set.Basic
```

Reply:
I have moved (and renamed) the definition to a separate file `Data.Set.FunctorToTypes`.